### DISCUSSION TITLE
Does Blaze support Transient objects

### DISCUSSION THREAD

**Discussion by rclarke1** [Blazebit/blaze-persistence#1496] [Q&A]
Hi, 
I am want to Set some custom objects into an EntityView that I have cached that are not part of a query, is this possible? I have tried annotating with Transient but that does is not working.
Thanks

**Comment by beikov** [comment]
Hello. You can use abstract classes and cache whatever you want, or even use `@MappingParameter` to inject non-existing optional parameters, which you could use to store data also within interfaces.

**Reply by rclarke1** [reply]
Thanks,
Can the @MappingParameter be used for a collection?
:  I am trying:
```
        @MappingParameter("serviceCodes")
	List<ServiceCodeDto> getServiceCodes(); 	
	void setServiceCodes(List<ServiceCodeDto>  set);
```

This working fine for me in on my code locally using Spring boot but when I push the code to heroku I get:

`The initialization of the static metamodel class 'nz.co.submitkit.blazeviews.AppointmentView_' failed: java.lang.IllegalArgumentException: Can not set static com.blazebit.persistence.view.metamodel.MethodListAttribute field nz.co.submitkit.blazeviews.AppointmentView_.serviceCodes to com.blazebit.persistence.view.impl.metamodel.attribute.MappingMethodSingularAttribute`

**Reply by rclarke1** [reply]
Wrapping the collection in another class worked for my purposes

**Reply by beikov** [reply]
It should be usable, but I guess this is a bug. Thanks for reporting. I created https://github.com/Blazebit/blaze-persistence/issues/1497 for this purpose
